### PR TITLE
P1718R2 Mandating the Standard Library: Clause 25 - Algorithms library

### LINKED ISSUE #3426: [2019-11 LWG Motion 24]  P1718R2 Mandating the Standard Library: Clause 25 - Algorithms library

### PR DESCRIPTION
**Pull request by zygoloid**
Fixes #3426.
Fixes cplusplus/papers#484.

### CODE REVIEW

File: source/algorithms.tex [hunk 1]
```diff
@@ -9681,19 +9755,20 @@
 
 \begin{itemdescr}
 \pnum
-\requires
+\mandates
+\tcode{M} are \tcode{N} both are integer types and not
```
Comment (johelegp) [inline source/algorithms.tex]:
```suggestion
\tcode{M} and \tcode{N} are both are integer types and not
```

File: source/algorithms.tex [hunk 2]
```diff
@@ -9715,16 +9790,17 @@
...
```
Comment (johelegp) [inline source/algorithms.tex]:
```suggestion
\tcode{M} and \tcode{N} are both are integer types and not
```

File: source/algorithms.tex [hunk 3]
```diff
@@ -8905,18 +8944,26 @@
...
-\requires
+\mandates
+All of
+  \begin{itemize}
+  \item \tcode{binary_op(init, *first)},
+  \item \tcode{binary_op(*first, init)},
+  \item \tcode{binary_op(init, init)}, and
+  \item \tcode{binary_op(*first, *first)}
+  \end{itemize}
+  are convertible to \tcode{T}.
```
Comment (jensmaurer) [inline source/algorithms.tex]: Should this be out-dented to match "All of" above?

Comment (zygoloid) [reply source/algorithms.tex]: Yes, thanks.

**Pull request merged**